Add documentation for how to administer Read the Docs for documentation and update Plone Sphinx Theme configuration

stevepiercy opened the pull request:
Closes https://github.com/plone/plone-sphinx-theme/issues/10

Closes #1604

Depends on merge of https://github.com/plone/volto/pull/6030 because its `custom.css` file pollutes default styles, turning the `plone.api` docs into an unreadable mess, as shown in https://plone6--1667.org.readthedocs.build/plone.api/api/user.html.

stevepiercy committed:
Add documentation for how to administer Read the Docs for documentation

stevepiercy committed:
Remove subprojects mention

stevepiercy committed:
Remove Netlify configuration

stevepiercy committed:
Merge branch '6.0' into pull-request-preview

stevepiercy committed:
Update requirements files

stevepiercy committed:
Remove netlify.toml

stevepiercy committed:
Add RTD configuration

stevepiercy committed:
Update conf.py for new theme

stevepiercy committed:
Add configuration for pull request previews on RTD

stevepiercy committed:
Try using deps to pull in submodules and create symlinks

stevepiercy committed:
RTD cannot run multiple make targets

stevepiercy committed:
Remove obsolete Sphinx template `_templates/sections/header-article.html`

stevepiercy committed:
Purge Netlify

stevepiercy committed:
Attempt to make plone.api importable to Sphinx

stevepiercy committed:
Attempt to make plone.api importable to Sphinx

stevepiercy committed:
Attempt to make plone.api importable to Sphinx

stevepiercy committed:
Attempt to make plone.api importable to Sphinx

stevepiercy committed:
Attempt to make plone.api importable to Sphinx

stevepiercy committed:
Attempt to make plone.api importable to Sphinx

stevepiercy committed:
Attempt to make plone.api importable to Sphinx

stevepiercy committed:
Attempt to make plone.api importable to Sphinx

stevepiercy committed:
Attempt to make plone.api importable to Sphinx by bypassing Makefile

stevepiercy committed:
Attempt to make plone.api importable to Sphinx by bypassing Makefile

stevepiercy committed:
Attempt to make plone.api importable to Sphinx by using its system Python instead of the Python in the Makefile

stevepiercy committed:
Attempt to make plone.api importable to Sphinx by using its system Python instead of the Python in the Makefile

stevepiercy committed:
Fix link references

stevepiercy committed:
Relocate static files

stevepiercy committed:
Purge Netlify

stevepiercy committed:
Clean up `conf.py`

stevepiercy committed:
Fix link

stevepiercy committed:
Revert "Try using deps to pull in submodules and create symlinks"

This reverts commit c23aa55618600d9e144ca6f1e1ca8a8ce56c892d.

stevepiercy committed:
Merge branch '6.0' into pull-request-preview

jensens commented on docs/contributing/documentation/admins.md:
```diff
@@ -84,49 +84,42 @@ To make it easier for other contributors to work with your project, update the f
+1.  Click {guilabel}`Finish`.
+    Read the Docs will redirect you to the project details, and start building the docs.
+
+For most Plone projects, you will not want to Read the Docs to publish the `latest` or other specific versions.
+Plone projects currently self-host their official documentation.
```
Sorry, I do not get exactly what this two lines are supposed to tell me.

stevepiercy replied:
We already have a public version that gets indexed by search engines. We don't want that for pull request previews. I also realized this section was kind of incomplete in regards to activating and hiding versions. Here's a revision.

````suggestion
Many Plone projects currently self-host their official documentation at {doc}`/index`.
These projects get indexed by search engines.

For pull request previews, unsupported branches or versions, or other situations, you most likely do not want search engines to index your documentation.
In this case, you can create a custom {file}`robots.txt` file to discourage, but not absolutely prevent, search engine indexing.
You can configure Sphinx to copy the {file}`robots.txt` file.
However, if you want to have two versions of a {file}`robots.txt` file—say one that allows indexing of your official documentation and another that discourages indexing—you can configuration your automation to copy it into place.

```shell
cp some-path/block-robots.txt docs-root-path/robots.txt
```

You can also configure which branches to activate or hide through Read the Docs.

```{seealso}
-   [`robots.txt` support](https://docs.readthedocs.io/en/stable/reference/robots.html)
-   [Automation rules](https://docs.readthedocs.io/en/stable/automation-rules.html)
-   [Versions](https://docs.readthedocs.io/en/stable/versions.html)
-   [Managing versions automatically](https://docs.readthedocs.io/en/stable/guides/automation-rules.html)
```
````

stevepiercy replied:
@jensens I revised this section and improved the flow. Please let me know.

stevepiercy committed:
Rename headings and improve flow

stevepiercy committed:
Rewrite search engine indexing section
Add Cancel builds programmatically section
Add Cancel builds on a branch section

stevepiercy committed:
Merge branch '6.0' into pull-request-preview

stevepiercy committed:
Update configuration to build only on changes to docs files

stevepiercy committed:
Use commands instead

stevepiercy committed:
Merge branch '6.0' into pull-request-preview

stevepiercy committed:
Merge branch '6.0' into pull-request-preview

stevepiercy commented:
I broke out pull request preview stuff because we need it now, since we purged netlify.

stevepiercy committed:
Merge branch '6.0' into pull-request-preview

# Conflicts:
#	_docs/_static/custom.css
#	docs/conf.py
#	docs/install/manage-add-ons-packages.md
#	requirements-initial.txt
#	requirements.txt

stevepiercy committed:
- Sort items in .gitignore
- Remove obsolete documentation-custom.css file
- Merge `conf.py` with 6.0

stevepiercy committed:
- Remove obsolete custom.css file

stevepiercy committed:
Update comments in linkcheckignore

stevepiercy committed:
plone.api must be installed to build its API docs. Remove symlinks in `distclean` allows its installation.

stevepiercy committed:
Tidy up conf.py

stevepiercy committed:
This file's content was migrated elsewhere

stevepiercy committed:
Disable Edit this page, and add comments explaining that it does not support multiple repositories

stevepiercy committed:
Update documentation for build pull request previews.
